api/plugin_support.h

```diff
@@ -219,13 +219,71 @@ int mambo_register_function_cb(mambo_context *ctx, char *fn_name,
                                mambo_callback cb_pre, mambo_callback cb_post, int max_args);
 
 /* Memory management */
+/**
+ * @brief A wrapper of mmap to allocate memory space optimised to support the
+ * DBM use case.
+ *
+ * An mmap wrapper that allocates private to MAMBO memory space with read/write
+ * permissions. This function shoud be used instead of the regular libc malloc
+ * inside plugins.
+ *
+ * @param ctx The context that holds the plugin state. Currently unused.
+ * @param size The allocation size requested. Will round up to a multiple of
+ * page size due to calling mmap.
+ * @return Pointer to the mapped area or @c (void*)-1 on error.
+ */
 void *mambo_alloc(mambo_context *ctx, size_t size);
+
+/**
+ * @brief Frees memory space pointed by @c ptr which must have been returned by a
+ * previous call to mambo_alloc.
+ *
+ * Currently this function is just and empty wrapper.
+ *
+ * @param ctx The context that holds the plugin state. Currently unused.
+ * @param ptr Pointer to memory space requested for freeing. Currently unused.
+ */
 void mambo_free(mambo_context *ctx, void *ptr);
 
 /* Access plugin data */
+/**
+ * @brief Stores an address to a global structure, making data of that address
+ * accessible between callbacks of all threads running in MAMBO.
+ *
+ * @param ctx The context that holds the plugin state.
+ * @param data The pointer of the address to be stored.
```

Comment:
Should be either: "The address to be stored" or "The pointer to be stored" or ".... pointer **to** the address ...". Otherwise it'd imply data is pointer to pointer. In this case think about void* as uintptr_t.